temporalcli/commands.go

```diff
@@ -0,0 +1,400 @@
+		// If level is off, make noop logger
+		if c.LogLevel.Value == "off" {
+			cctx.Logger = newNopLogger()
+		} else {
+			var level slog.Level
+			if err := level.UnmarshalText([]byte(c.LogLevel.Value)); err != nil {
+				return fmt.Errorf("invalid log level %q: %w", c.LogLevel.Value, err)
+			}
+			var handler slog.Handler
+			switch c.LogFormat.Value {
+			case "text":
+				handler = slog.NewTextHandler(cctx.Options.Stderr, &slog.HandlerOptions{
+					Level: level,
+					// Remove the TZ
+					ReplaceAttr: func(groups []string, a slog.Attr) slog.Attr {
+						if a.Key == slog.TimeKey && a.Value.Kind() == slog.KindTime {
+							a.Value = slog.StringValue(a.Value.Time().Format("2006-01-02T15:04:05.000"))
+						}
+						return a
+					},
+				})
+			case "json":
+				handler = slog.NewJSONHandler(cctx.Options.Stderr, &slog.HandlerOptions{Level: level})
+			default:
+				return fmt.Errorf("invalid log format %q", c.LogFormat.Value)
+			}
+			cctx.Logger = slog.New(handler)
+		}
+	}
+
+	// Configure printer if not already on context
+	cctx.JSONOutput = c.Output.Value == "json"
+	if cctx.Printer == nil {
+		cctx.Printer = &printer.Printer{
+			Output:               cctx.Options.Stdout,
+			JSON:                 cctx.JSONOutput,
+			JSONIndent:           "  ",
+			JSONPayloadShorthand: !c.NoJsonShorthandPayloads,
+		}
+		switch c.TimeFormat.Value {
+		case "iso":
+			cctx.Printer.FormatTime = func(t time.Time) string { return t.Format(time.RFC3339) }
+		case "raw":
+			cctx.Printer.FormatTime = func(t time.Time) string { return fmt.Sprintf("%v", t) }
+		case "relative":
+			cctx.Printer.FormatTime = humanize.Time
+		default:
+			return fmt.Errorf("invalid time format %q", c.TimeFormat.Value)
+		}
+	}
+	cctx.JSONShorthandPayloads = !c.NoJsonShorthandPayloads
+	return nil
+}
+
+// May be empty result if can't get user home dir
+func defaultEnvConfigFile(appName, configName string) string {
+	// No env file if no $HOME
+	if dir, err := os.UserHomeDir(); err == nil {
+		return filepath.Join(dir, ".config", appName, configName+".yaml")
+	}
+	return ""
+}
+
+func readEnvConfigFile(file string) (env map[string]map[string]string, err error) {
+	b, err := os.ReadFile(file)
+	if os.IsNotExist(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, fmt.Errorf("failed reading env file: %w", err)
+	}
+	var m map[string]map[string]map[string]string
+	if err := yaml.Unmarshal(b, &m); err != nil {
+		return nil, fmt.Errorf("failed unmarshalling env YAML: %w", err)
+	}
+	return m["env"], nil
+}
+
+func writeEnvConfigFile(file string, env map[string]map[string]string) error {
+	b, err := yaml.Marshal(map[string]any{"env": env})
+	if err != nil {
+		return fmt.Errorf("failed marshaling YAML: %w", err)
+	}
+	// Make parent directories as needed
+	if err := os.MkdirAll(filepath.Dir(file), 0755); err != nil {
+		return fmt.Errorf("failed making env file parent dirs: %w", err)
+	} else if err := os.WriteFile(file, b, 0644); err != nil {
```

Comment:
:+1: Will change. There is a known quirk w/ `MkdirAll` where it uses this for all dirs, so if the created grandparent doesn't give privs to create the parent it will fail, but I think `0700` should still be fine for this case.

Reply:
Updated